Tests/SwiftDocCUtilitiesTests/ConvertActionTests.swift

```diff
@@ -1664,6 +1664,89 @@ class ConvertActionTests: XCTestCase {
+        // which is required for testing if the conversion output is deterministic.
+        let priorPrettyPrintValue = shouldPrettyPrintOutputJSON
+        shouldPrettyPrintOutputJSON = true
+        defer {
+            shouldPrettyPrintOutputJSON = priorPrettyPrintValue
+        }
+        
+        let testBundleURL = try XCTUnwrap(
+            Bundle.module.url(
+                forResource: "TestBundle",
+                withExtension: "docc",
+                subdirectory: "Test Bundles"
+            )
+        )
+        let bundle = try Folder.createFromDisk(url: testBundleURL)
+        
+        func performConvertAction(outputURL: URL, testFileSystem: TestFileSystem) throws {
+            var action = try ConvertAction(
+                documentationBundleURL: bundle.absoluteURL,
+                outOfProcessResolver: nil,
+                analyze: false,
+                targetDirectory: outputURL,
+                htmlTemplateDirectory: Folder.emptyHTMLTemplateDirectory.absoluteURL,
+                emitDigest: false,
+                currentPlatforms: nil,
+                dataProvider: testFileSystem,
+                fileManager: testFileSystem
+            )
+            
+            _ = try action.perform(logHandle: .none)
+        }
+        
+        // We'll perform 3 sets of conversions to confirm the output is deterministic
+        for _ in 1...3 {
+            let testFileSystem = try TestFileSystem(
+                folders: [bundle, Folder.emptyHTMLTemplateDirectory]
+            )
+            
+            // Convert the same bundle three times and place the output in
+            // separate directories.
+            
+            try performConvertAction(
+                outputURL: URL(fileURLWithPath: "/1", isDirectory: true),
+                testFileSystem: testFileSystem
+            )
+            try performConvertAction(
+                outputURL: URL(fileURLWithPath: "/2", isDirectory: true),
+                testFileSystem: testFileSystem
+            )
+            
+            // Extract and sort the RenderJSON output of each conversion
+            
+            let firstConversionFiles = testFileSystem.files.lazy.filter { key, _ in
+                key.hasPrefix("/1/data/")
+            }.map { (key, value) in
+                return (String(key.dropFirst(2)), value)
+            }.sorted(by: \.0)
+            
+            let secondConversionFiles = testFileSystem.files.lazy.filter { key, _ in
+                key.hasPrefix("/2/data/")
+            }.map { (key, value) in
+                return (String(key.dropFirst(2)), value)
+            }.sorted(by: \.0)
+            
+            // Zip the two sets of sorted files and loop through them, ensuring that
+            // each conversion produced the same RenderJSON output.
+            
+            for (first, second) in zip(firstConversionFiles, secondConversionFiles) {
```

Comment:
We should check that `firstConversionFiles.count` == `secondConversionFiles.count` because `zip` will silently drop unmatched pairs if the sequences have different lengths.

Reply:
Ah good catch. I went ahead and added an `XCTAssertEqual` above that verifies that each array of file paths are entirely equal. This should address that issue and allows us to remove the individual assert in this loop.